client/src/partials/templates/dateEditor.tmpl.html

```diff
@@ -0,0 +1,40 @@
+<!-- Date editor component view implementation-->
+<div 
+	ng-switch="$ctrl.editMode"
+	class="form-group"
+	ng-class="{'has-error' : $ctrl.validationTrigger && $ctrl.form.dateEditorValue.$invalid}">
+
+	<!-- Input component implementation-->
+	<div class="input-group">
+	    <input
+	    	data-date-editor
+	    	name="dateEditorValue"
+	    	type="text"
+	    	class="form-control"	    	
+	    	uib-datepicker-popup="{{$ctrl.dateFormat}}"
+	    	ng-model="$ctrl.dateValue"
+	    	is-open="$ctrl.editMode"
+	    	min-date="$ctrl.minDate"
+	    	max-date="$ctrl.maxDate"
+	    	datepicker-options="dateOptions"
+	    	close-text="Close"
+	    	alt-input-formats="altInputFormats"
+	    	ng-disabled="!$ctrl.editMode"
+	    	show-button-bar="true"
+	    	required
+	    	/>
+	    <span class="input-group-btn">
+	    	<button style="width:8em;" class="btn btn-default" type="button" ng-click="$ctrl.changeMode()" ng-disabled="$ctrl.editMode">
+	    		{{"DATEFINDER.SET" | translate}}
+	    	</button>
+	    </span>
+    </div> 
+
+    <!-- Error message implementation-->
+    <p class="help-block" ng-messages="$ctrl.form.dateEditorValue.$error" ng-show="$ctrl.validationTrigger">
```

Comment:
Could you include error messages for `minDate` and `maxDate` errors as well?  It would also be nice to have a warning (not an error) for dates in the future.  If necessary, this could be behind a flag.